mne/time_frequency/tfr.py

```diff
@@ -245,6 +248,244 @@ def _cwt(X, Ws, mode="same", decim=1, use_fft=True):
+        The frequencies.
+    sfreq : float | int, defaults to 1.0
+        Sampling frequency of the data.
+    method : 'mtm' | 'morlet', defaults to 'morlet'
+        The time frequency method. 'morlet' convolves a Morlet wavelet. 'mtm'
+        convolves DPSS tapers.
+    n_cycles : float | array of float, defaults to 7.0
+        Number of cycles  in the Morlet wavelet. Fixed number
+        or one per frequency.
+    zero_mean : bool | None, defaults to None, which will use True for method='mtm' and False for method='morlet'. # noqa
+        Make sure the wavelets have a mean of zero.
+        method == 'mtm' and defaults to False if method == 'morlet'.
+    time_bandwidth : float, defaults to 4.0 (3 tapers)
+        Time x (Full) Bandwidth product. Only applies if method == 'mtm'.
+        The number of good tapers (low-bias) is chosen automatically based on
+        this to equal floor(time_bandwidth - 1).
+    use_fft : bool, defaults to True
+        Use the FFT for convolutions or not.
+    decim : int | slice, defaults to 1
+        To reduce memory usage, decimation factor after time frequency
+        decomposition.
+        If `int`, returns tfr[..., ::decim].
+        If `slice` returns tfr[..., decim].
+        .. note: Decimation may create aliasing artifacts.
+    output : str, defaults to 'complex'
+
+        * 'complex' : single trial complex.
+        * 'power' : single trial power.
+        * 'phase' : single trial phase.
+        * 'avg_power' : average of single trial power.
+        * 'itc' : inter-trial coherence.
+        * 'avg_power_itc' : average of single trial power and inter-trial
+          coherence across trials.
+
+    n_jobs : int, defaults to 1
+        The number of epochs to process at the same time. The parallization is
+        implemented across channels.
+    verbose : bool, str, int, or None, defaults to None
+        If not None, override default verbose level (see mne.verbose).
+
+    Returns
+    -------
+    out : array
+        Time frequency transform of epoch_data. If output is in ['complex',
+        'phase', 'power'], then shape of out is (n_epochs, n_chans, n_freqs,
+        n_times), else it is (n_chans, n_freqs, n_times). If output is
+        'avg_power_itc', the real values code for 'avg_power' and the
+        imaginary values code for the 'itc': out = avg_power + i * itc
+    """
+    # Check data
+    epoch_data = np.asarray(epoch_data)
+    if epoch_data.ndim != 3:
+        raise ValueError('epoch_data must be of shape '
+                         '(n_epochs, n_chans, n_times)')
+
+    # Check frequencies
+    if not isinstance(frequencies, (list, np.ndarray)):
+        raise ValueError('frequencies must be an array-like.')
+    frequencies = np.asarray(frequencies, dtype=float)
+    if frequencies.ndim != 1:
+        raise ValueError('frequencies must be of shape n_freqs.')
+
+    # Check sfreq
+    if not isinstance(sfreq, (float, int)):
```

Comment:
this can crash in python 3.5 for numpy types, to be kept in mind. Another test could be `np.isreal`. Probably it would be unlikely not to pass pure python types here but in other places it actually can pose problems.